filter.js

```diff
@@ -0,0 +1,168 @@
+/*------------------------FILTER VALUES------------------------*/
+
+let filter = {
+    online: true, //online challenges checkbox
+    onsite: true, //onsite challenges checkbox
+    minRating: 1, //lowest rating
+    maxRating: 5, //highest rating
+    labels: [], //active tags in lowercase
+    words: [], //inputfield words split into strings in lowercase
+  };
+
+  export default filter;
+  
+  /*------------------------FILTER VALUES------------------------*/
+  
+  /*------------------------BY TYPE------------------------*/
+  
+  // Get all the types
+  const online = document.querySelector("#online");
+  const onsite = document.querySelector("#onsite");
+  
+  // Functions to handle type click
+  online.addEventListener("click", function () {
+    if (filter.online == true) {
+      filter.online = false;
+    } else {
+      filter.online = true;
+    }
+  });
+  
+  onsite.addEventListener("click", function () {
+    if (filter.onsite == true) {
+      filter.onsite = false;
+    } else {
+      filter.onsite = true;
+    }
+  });
+  
+  /*------------------------BY TYPE------------------------*/
+  
+  /*------------------------STAR RATING------------------------*/
+  
+  // Get all the stars
+  const fromStars = [
+    document.querySelector("#fromOne"),
+    document.querySelector("#fromTwo"),
+    document.querySelector("#fromThree"),
+    document.querySelector("#fromFour"),
+    document.querySelector("#fromFive"),
+  ];
+  
+  const toStars = [
+    document.querySelector("#toOne"),
+    document.querySelector("#toTwo"),
+    document.querySelector("#toThree"),
+    document.querySelector("#toFour"),
+    document.querySelector("#toFive"),
+  ];
+  
+  // Add event listeners to all stars
+  fromStars.forEach((star, index) => {
+    star.addEventListener("click", () => handleStarClick(fromStars, index, true));
+  });
+  
+  toStars.forEach((star, index) => {
+    star.addEventListener("click", () => handleStarClick(toStars, index, false));
+  });
+  
+  // Function to handle star click
+  function handleStarClick(stars, index, isFromStars) {
+    // Remove 'checked' class from all stars
+    stars.forEach((star) => star.classList.remove("checked"));
+  
+    // Add 'checked' class to clicked star and all previous stars
+    for (let i = 0; i <= index; i++) {
+      stars[i].classList.add("checked");
+    }
+  
+    // Update the rating
+    const rating = index + 1;
+    if (isFromStars) {
+      // Only update minRating if it's less than or equal to maxRating
+      if (rating <= filter.maxRating) {
+        filter.minRating = rating;
+      } else {
+        // If minRating would be greater than maxRating, remove 'checked' class from the stars that should not be selected
+        for (let i = filter.maxRating; i <= index; i++) {
+          stars[i].classList.remove("checked");
+        }
```

Comment:
Nice feature! It would have been awesome if it worked the other way around as well (i.e. automatically updating the max rating when min rating is changed to a value greater than max rating).